File: backend/routes/leave_request.go
package routes

import (
	"oa-system/handlers"

	"github.com/gin-gonic/gin"
)

// leaveRequestRoutes 执行相关业务逻辑
func leaveRequestRoutes(rg *gin.RouterGroup) {
	rg.GET("/leave-requests", handlers.GetLeaveRequests)
	rg.GET("/leave-requests/:id", handlers.GetLeaveRequest)
	rg.POST("/leave-requests", handlers.CreateLeaveRequest)
	rg.PUT("/leave-requests/:id", handlers.UpdateLeaveRequest)
	rg.PUT("/leave-requests/:id/submit", handlers.SubmitLeaveRequest)
	rg.PUT("/leave-requests/:id/approve", handlers.ApproveLeaveRequest)
	rg.DELETE("/leave-requests/:id", handlers.DeleteLeaveRequest)
}

File: frontend/src/api/leave_request.js
import http from './http'

export const getLeaveRequests = (params) => http.get('/leave-requests', { params })
export const getLeaveRequest = (id) => http.get(`/leave-requests/${id}`)
export const createLeaveRequest = (data) => http.post('/leave-requests', data)
export const updateLeaveRequest = (id, data) => http.put(`/leave-requests/${id}`, data)
export const submitLeaveRequest = (id, data) => http.put(`/leave-requests/${id}/submit`, data)
export const approveLeaveRequest = (id, data) => http.put(`/leave-requests/${id}/approve`, data)
export const deleteLeaveRequest = (id) => http.delete(`/leave-requests/${id}`)
